src/foam/nanos/crunch/ui/UserCapabilityJunctionWAO.js

```diff
@@ -18,15 +18,26 @@ foam.CLASS({
     'crunchService'
   ],
 
+  properties: [
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.auth.Subject',
+      name: 'subject'
+    }
+  ],
+
   methods: [
     function save(wizardlet) {
       if ( ! wizardlet.isAvailable ) return Promise.resolve();
+      var wData = wizardlet.data ? wizardlet.data.clone() : null;
       wizardlet.loading = true;
-      return this.crunchService.updateJunction( null,
-        wizardlet.capability.id,
-        wizardlet.data ? wizardlet.data.clone() : null,
-        null
-      ).then((ucj) => {
+      let p = this.subject ? this.crunchService.updateJunctionFor(
```

Comment:
when would the subject exist and no exist?
I'd recommend a comment - not specifically mine - this is just a suggetion
```suggestion
      // Generic model used in backoffice & user application. Subject ove-ride should only be hit if backoffice
      let p = this.subject ? this.crunchService.updateJunctionFor(
```

Reply:
Added documentation to subject property for clarification.